sanic_routing/route.py

```diff
@@ -84,21 +92,38 @@ def __str__(self):
         )
         return f"<{self.__class__.__name__}: {display}>"
 
+    def __repr__(self) -> str:
+        return str(self)
+
     def __eq__(self, other) -> bool:
         if not isinstance(other, self.__class__):
             return False
+
+        # Equality specifically uses self.segments and not self.parts.
+        # In general, these properties are nearly identical.
+        # self.segments is generalized and only displays dynamic param types
+        # and self.parts has both the param key and the param type
         return bool(
             (
-                self.parts,
+                self.segments,
                 self.requirements,
             )
             == (
-                other.parts,
+                other.segments,
                 other.requirements,
             )
             and (self.methods & other.methods)
```

Comment:
If *any* methods match, there is equality. If this is intended (rather than requiring identical methods), a comment would be useful.

Reply:
Yes, this is the intention.

One could be `GET,PUT,POST` and the other is `GET`: we want them to match.

Comment in next push.